### ISSUE TITLE
Data-Binding to updated MaterializeParams from pickadate not working

### ISSUE THREAD

**Issue by GregorBiswanger** [InfomediaLtd/angular2-materialize#127]
Hi,
I set a object property for MaterializeParams.. on load the pickadate control works..
But if I changed the object to runtime, the pickadate control dont get the new param values..
Do you have any idea what is wrong or can you fix the bug please.

Here my view:
```
<input id="from" formControlName="fromDate" type="text" class="datepicker" materialize="pickadate" [materializeParams]="[fromDateOptions]">  
...
<input id="to" formControlName="toDate" type="text" class="datepicker" materialize="pickadate" [materializeParams]="[toDateOptions]">
```

Here my component code:
```
...
  fromDateOptions = this.getDefaultPickaOption();
...
  private getDefaultPickaOption(): Pickadate.DateOptions {
    return {
      labelMonthNext: 'Nächster Monat',
      labelMonthPrev: 'Letzter Monat',
      labelMonthSelect: 'Monat auswählen',
      labelYearSelect: 'Jahr auswählen',
      monthsFull: ['Januar', 'Februar', 'März', 'April', 'Mai', 'Juni', 'Juli', 'August', 'September', 'Oktober', 'November', 'Dezember'],
      monthsShort: ['Jan', 'Feb', 'Mär', 'Apr', 'Mar', 'Jun', 'Jul', 'Aug', 'Sep', 'Okt', 'Nov', 'Dez'],
      weekdaysFull: ['Sonntag', 'Montag', 'Dienstag', 'Mittwoch', 'Donnerstag', 'Freitag', 'Samstag'],
      weekdaysShort: ['So', 'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa'],
      today: 'Heute',
      clear: 'LÖSCHEN',
      close: 'OK',
      format: 'dd.mm.yyyy',
      closeOnSelect: true
    };
  }
```
...
later to runtime I changed the value:

```
  this.reportForm.controls['fromDate'].valueChanges.subscribe((value) => {
      this.toDateOptions.min = value;
      this.reportForm.controls['toDate'].setValue('');
    });

```

On the initialization the controls get the params, but changes on runtime dosn´t work...

Thanks

**Comment by krustypop** [comment]
Hi,

I have the same issue, when i try to update the 'min' option, this doesn't work. Is anybody would have a workaround ?
Do i have to destroy the component with jQuery to reload the datepicker with new options ?

Thanks

**Comment by AAAnderson7301** [comment]
@GregorBiswanger How do you get access to Pickadate ?

**Comment by GregorBiswanger** [comment]
I use the Angular2 ReactiveFormsModule with Observable (RxJS):

```
this.reportForm.controls['fromDate'].valueChanges.subscribe((value) => {
      this.toDateOptions.min = value;
      this.reportForm.controls['toDate'].setValue('');
    });
```

**Comment by AAAnderson7301** [comment]
Hi @GregorBiswanger Sorry I was not clear. I am using reactive forms as well I mean in the component when you specify :Pickadate.DateOptions()
Are you importing it?

**Comment by GregorBiswanger** [comment]
No, I have only my date options object on component and bind it with materializeParams...

**Comment by AAAnderson7301** [comment]
I am just curious as to how this line works. ```private getDefaultPickaOption(): Pickadate.DateOptions {``` 
I am getting "cannot find namespace 'Pickadate'" You are saying that getDefaultPickaOption is of type Pickadate so I am just trying to figure out how you can set the type

**Comment by GregorBiswanger** [comment]
Ahh okay.. I understand you now :)
That comes from the type definition for Pickadate.. Materialize use the jQuery Pickadate options...
`npm install --save @types/pickadate`

_@types is only supported from TypeScript 2._

**Comment by AAAnderson7301** [comment]
Awesome thank you. I was trying to do it like this and pass it in from the html page but only the format works ```[materializeParams]="[{format:'yyyy-mm-dd', weekdaysShort: ['So', 'Mo', 'Di', 'Mi', 'Do', 'Fr', 'Sa']}]"```

**Comment by GregorBiswanger** [comment]
It works fine for me.. I only have the problem to change the params a second time.. the control doesn't get the new params...

**Comment by rubyboy** [comment]
@GregorBiswanger try ```materializeActions``` to force the component to refresh its state.

Another thing to try is to assign a whole new value to your toDateOptions (a-la ```this.toDateOptions = Object.assign({}, this.toDateOptions, {min:value})```). This will create a new reference and should kick off Angular's change detection and hopefully refresh the component.

**Comment by GregorBiswanger** [comment]
@rubyboy How can I see, what actions are available?

**Comment by rubyboy** [comment]
Also discussed here: https://github.com/InfomediaLtd/angular2-materialize/issues/280

**Comment by diandsonc** [comment]
Solved here [https://github.com/InfomediaLtd/angular2-materialize/issues/280](#280)
